dragonchain/lib/broadcast.py

```diff
@@ -100,7 +118,11 @@ def send_receipts(l5_block: "l5_block_model.L5BlockModel") -> None:
                     claim_check_id = f"{chain_id}-{block}"
                     matchmaking.resolve_claim_check(claim_check_id)
                 except Exception:
-                    _log.exception("Failure to finalize claim in matchmaking. Sending reciepts to lower level nodes.")
+                    _log.exception(
+                        "Failure to finalize claim in matchmaking. Sending receipts to lower level nodes. \n"
+                        f"Adding claim to failed queue.  Claim ID: {claim_check_id}"
+                    )
+                    redis.lpush_sync("mq:failed-claims", claim_check_id)
```

Comment:
this needs to be a confirmed retry-able exception. We can't just blindly retry.

For example, if we got back an HTTP 400 from matchmaking, we should NOT retry, and should rather raise the exception since there is NO reason an L5 should ever get a 400 from matchmaking, and if it did, something is wrong, and we definitely shouldn't retry.

Reply:
Created exception type for 500-level errors and restricted queue additions to these exceptions only.